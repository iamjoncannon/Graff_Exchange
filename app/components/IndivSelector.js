import React from 'react';
import { Link } from 'react-router-dom'

export default function IndivSelector (props){

  const { pathname } = props;

  function isSelectedSelector(selector){

    return pathname.includes(selector) ? "selected" : undefined ;
  }

  return (

    <div onClick={props.exit}>

      <Link to="/indiv/perf">

        <span className={isSelectedSelector("perf")}>
          Performance
        </span>          
      
      </Link>

      <Link to="/indiv/finan">

        <span className={isSelectedSelector("finan")}>
          Financials
        </span>

      </Link>

      <Link to="/indiv/news">
      
        <span className={isSelectedSelector("news")}>
          News
        </span>
      
      </Link>

      <Link to="/indiv/trans">
      
        <span className={isSelectedSelector("trans")}>
          Transactions
        </span>
      
      </Link>
    </div>
    
  );
};

