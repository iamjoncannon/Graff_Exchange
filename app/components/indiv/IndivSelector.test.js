import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import IndivSelector from './IndivSelector'
import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"
const mockStore = configureStore([thunk]);
import { BrowserRouter as Router } from "react-router-dom";

describe("IndivSelector", ()=>{

    let component 
    let store 

    beforeEach(()=>{
                
        store = mockStore( );
    
        component = renderer.create(
            <Provider store={store} >
               <Router >
                   <IndivSelector pathname={"/"} />
               </Router>
           </Provider>)
    })

    it('1) renders successfully with data from redux store', ()=>{

        let tree = component.toJSON();
        
        expect(tree).toMatchSnapshot();
    })

    it("2) renders four links with span children", ()=>{

        const links = component.root.findAll(node=> node.type.displayName === "Link")

        expect(links.length).toEqual(4)

        const spans = component.root.findAllByType("span")

        expect(spans.length).toEqual(4)
    })
})

describe("IndivSelector", ()=>{

    let component 
    let store 

    beforeEach(()=>{
                
        store = mockStore( );
    
        component = renderer.create(
            <Provider store={store} >
               <Router >
                   <IndivSelector pathname={"/perf"} />
               </Router>
           </Provider>)
    })

    it("3) if pathname includes abbreviation, className is 'selected' ", ()=>{

        const selected = component.root.find(node => node.props.className === "selected")

        console.log(selected.children)

        expect(selected.children[0]).toEqual(Performance)
    })
})
