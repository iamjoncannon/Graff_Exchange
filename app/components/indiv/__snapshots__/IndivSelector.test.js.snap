// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndivSelector 1) renders successfully with data from redux store 1`] = `
<div>
  <a
    href="/indiv/perf"
    onClick={[Function]}
  >
    <span>
      Performance
    </span>
  </a>
  <a
    href="/indiv/finan"
    onClick={[Function]}
  >
    <span>
      Financials
    </span>
  </a>
  <a
    href="/indiv/news"
    onClick={[Function]}
  >
    <span>
      News
    </span>
  </a>
  <a
    href="/indiv/trans"
    onClick={[Function]}
  >
    <span>
      Transactions
    </span>
  </a>
</div>
`;
